### PR TITLE
API: improve swagger and go-doc for ImageInspect structs

### PR DESCRIPTION
**Pull request by thaJeztah**
- ~addresses https://github.com/docker/cli/issues/2868~
- rename definition in swagger from `Image` to `ImageInspect` to match the go type
- improve (or add) documentation for various fields
- move example values in-line in the "definitions" section
- remove the `required` fields from `ImageInspect`, as the type is only used as
  response type (not to make requests).

also improve go-doc for some structs accordingly, and update API docs for v1.39-v1.41 with the same changes

### CODE REVIEW

File: api/types/types.go [hunk 1]
```diff
@@ -27,30 +27,108 @@ type RootFS struct {
...
+	Architecture string
+
+	// Variant is the CPU architecture variant (presently ARM-only).
+	Variant string `json:",omitempty"`
+
+	// OS is the Operating System used to build and run the image.
+	Os string
+
+	// OsVersion is the Operating System version used to build and run the image.
+	OsVersion string `json:",omitempty"`
```
Comment (tianon) [inline api/types/types.go:99]: These aren't quite right -- I mean, they're correct, but incidentally, not that these are what the field means? :sweat_smile:

For example, it's conceivably possible to build an `{"os":"freebsd"}` image entirely on Linux, and the `os.version` is really the version of `Os` the image targets.

Maybe something like this instead?

```suggestion
	// OS is the Operating System the image is built to run on.
	Os string

	// OsVersion is the version of the Operating System the image is built to run on (especially for Windows).
	OsVersion string `json:",omitempty"`
```

File: api/types/types.go [hunk 2]
```diff
@@ -27,30 +27,108 @@ type RootFS struct {
...
+	// Depending on how the image was created, this field may be empty.
+	DockerVersion string
+
+	// Author is the name of the author that was specified when committing the
+	// image, or as specified through MAINTAINER (deprecated) in the Dockerfile.
+	Author string
+	Config *container.Config
+
+	// Architecture is the hardware CPU architecture that the image is built and
+	// runs on.
```
Comment (tianon) [inline api/types/types.go:89]: Similarly to `Os`, there's even less reason for me to have to use the "right" architecture to _build_ images. :sweat_smile:

```suggestion
	// Architecture is the hardware CPU architecture that the image runs on.
```

File: image/image.go [hunk 3]
```diff
@@ -36,23 +36,31 @@ func IDFromDigest(digest digest.Digest) ID {
...
+	// Depending on how the image was created, this field may be empty.
 	DockerVersion string `json:"docker_version,omitempty"`
-	// Author is the name of the author that was specified when committing the image
+	// Author is the name of the author that was specified when committing the
+	// image, or as specified through MAINTAINER (deprecated) in the Dockerfile.
 	Author string `json:"author,omitempty"`
 	// Config is the configuration of the container received from the client
 	Config *container.Config `json:"config,omitempty"`
-	// Architecture is the hardware that the image is built and runs on
+	// Architecture is the hardware CPU architecture that the image is built and runs on.
```
Comment (tianon) [inline image/image.go:63]: Doh, now I see -- I guess this is where that "built and runs on" verbiage came from? :grimacing: :see_no_evil: :innocent:

**Review by tianon — APPROVED** [review]
Overall I like it, just a few nits on the specific terminology added to the godocs, which I realized later might not actually be _new_ verbiage (rather copied from elsewhere)?

**Comment by thaJeztah** [comment]
@tianon updated, PTAL ❤️

**Review by tianon — APPROVED** [review]
:+1: :heart:

**Comment by thaJeztah** [comment]
Bringing this one in; I have a couple more follow-up changes in the swagger file(s)

**Pull request merged**